app/src/main/java/emu/skyline/AppDialog.kt

```diff
@@ -117,4 +230,9 @@ class AppDialog : BottomSheetDialogFragment() {
             }
         }
     }
+
+    override fun onDestroyView(){
+        super.onDestroyView()
+        lastZipCreated?.delete()
```

Comment:
Don't keep a zip file laying around in the `savesFolderPath`, but use the app cache folder, in the same way the import function does. That way you don't need to destroy the file here, but you'd destroy it right after "returning" from the intent. See `startActivityForResult`.